git_machete/cli.py

```diff
@@ -615,25 +617,29 @@ def launch(orig_args: List[str]) -> None:
             github_subcommand = parsed_cli.subcommand
             machete_client.read_definition_file()
 
-            if 'draft' in parsed_cli and github_subcommand not in {'create-pr'}:
-                raise MacheteException(
-                    "'--draft' option is only valid with 'create-pr' subcommand.")
-            if 'pr_no' in parsed_cli and github_subcommand not in {'checkout-prs'}:
-                raise MacheteException(
-                    "'pr_no' argument is only valid with 'checkout-prs' subcommand.")
+            if 'draft' in parsed_cli and github_subcommand != 'create-pr':
+                raise MacheteException("'--draft' option is only valid with 'create-pr' subcommand.")
+            for command in ('all', 'by', 'mine'):
+                if command in parsed_cli and github_subcommand != 'checkout-prs':
+                    raise MacheteException(f"'--{command}' argument is only valid with 'checkout-prs' subcommand.")
+            if 'pr_no' in parsed_cli and github_subcommand != 'checkout-prs':
+                raise MacheteException("'pr_no' option is only valid with 'checkout-prs' subcommand.")
 
             if github_subcommand == "anno-prs":
                 machete_client.sync_annotations_to_github_prs()
             elif github_subcommand == "checkout-prs":
-                if 'pr_no' not in parsed_cli:
+                if len(parsed_cli_as_dict) > 3 or len(parsed_cli_as_dict) == 2:
```

Comment:
Hmm just
```suggestion
                if len(parsed_cli_as_dict) > 2:
```

?

Reply:
I dont think so:
before: {2} U (3, +inf)
yours: (2, +inf)
value=3 should be allowed but in your version it's not.